libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/weChatSchema.ts

```diff
@@ -0,0 +1,1524 @@
+        this.MediaId = mediaId;
+        this.Title = title;
+        this.Description = description;
+    }
+}
+
+export class VideoResponse extends ResponseMessage {
+    Video: Video;
+    constructor(mediaId: string, title?: string, description?: string) {
+        super();
+        this.Video = new Video(mediaId, title, description);
+        this.MsgType = ResponseMessageTypes.Video;
+    }
+}
+
+export class Voice {
+    MediaId: string;
+    constructor(mediaId: string) {
+        this.MediaId = mediaId;
+    }
+}
+
+export class VoiceResponse extends ResponseMessage {
+    Voice: Voice;
+    constructor(mediaId: string) {
+        super();
+        this.Voice = new Voice(mediaId);
+        this.MsgType = ResponseMessageTypes.Voice;
+    }
+}
+
+export class MessageMenuResponse extends ResponseMessage {
+    MessageMenu: MessageMenu;
+
+    MsgType: ResponseMessageTypes.MessageMenu;
+}
+
+export interface MenuItem {
+    /**
+     * Gets or sets Id.
+     * @value
+     * Id of the menu item.
+     */
+    id: string;
+
+    /**
+     * Gets or sets Content.
+     * @value
+     * Content of the menu item.
+     */
+    content: string;
+}
+
+export interface MessageMenu {
+    /**
+     * Gets or sets HeaderContent.
+     * @value
+     * HeaderContent of the menu.
+     */
+    HeaderContent: string;
+
+    /**
+     * Gets or sets MenuItems.
+     * @value
+     * Items in message menu.
+     */
+    MenuItems: MenuItem[];
+
+    /**
+     * Gets or sets TailContent.
+     * @value
+     * Footer of the menu.
+     */
+    TailContent: string;
+}
+
+/**Secret info store the parameter used to verify the message from WeChat and decrypt message content. */
+export interface SecretInfo {
+    /**
+     * Gets or Sets signature from WeChat update webhook request.
+     * @value
+     * signature from WeChat update webhook request.
+     */
+    Signature: string;
+
+    /**
+     * Gets or Sets signature from WeChat message request.
+     * @value
+     * Signature from WeChat message request.
+     */
+    Msg_signature: string;
+
+    /**
+     * Gets or Sets timestamp.
+     * @value
+     * Timestamp of the request parameter.
+     */
+    Timestamp: string;
+
+    /**
+     * Gets or Sets nonce.
+     * @value
+     * Nonce of the request parameter.
+     */
+    Nonce: string;
+
+    /**
+     * Gets or Sets token.
+     * @value
+     * Token from the request parameter.
+     */
+    Token: string;
+
+    /**
+     * Gets or Sets endcoding aes key.
+     * @value
+     * EncodingAESKey from appsetings.
+     * EncodingAESKey fixed length of 43 characters, a-z, A-Z, 0-9 a total of 62 characters selected
+     * https://open.weixin.qq.com/cgi-bin/showdocument?action=dir_list&t=resource/res_list&verify=1&id=open1419318479&token=&lang=en_US.
+     */
+    EncodingAesKey: string;
+
+    /**
+     * Gets or Sets WeChat app id.
+     * @value
+     * WeChat app id.
+     */
+    AppId: string;
+}
+
+export class WeChatAccessToken {
```

Comment:
>export class WeChatAccessToken [](start = 0, length = 30)

This should be an interface... Classes won't round trip when serialized to storage.